### PR TITLE
Prohibit `sk=0` at key import

### PR DESCRIPTION
**Pull request by survived**

### CODE REVIEW

**Review by maurges — DISMISSED** [review]
Damn cargo.lock is thicc

File: Cargo.toml [hunk 1]
```diff
@@ -6,4 +6,11 @@ members = [
...
-]
\ No newline at end of file
+]
+
+[patch.crates-io.generic-ec]
+git = "https://github.com/dfns/generic-ec"
+branch = "nonzero-cmp"
+[patch.crates-io.generic-ec-zkp]
+git = "https://github.com/dfns/generic-ec"
+branch = "nonzero-cmp"
```
Comment (survived) [inline Cargo.toml]: Need to merge that first

File: Cargo.toml [hunk 2]
```diff
@@ -6,4 +6,8 @@ members = [
   "./key-export-common",
   "./key-export-client",
   "./core",
-]
\ No newline at end of file
+]
+
+[patch.crates-io.generic-ec]
+git = "https://github.com/dfns/generic-ec"
+branch = "m"
```
Comment (survived) [inline Cargo.toml]: Do we want to publish new version of generic-ec first?

**Comment by survived** [comment]
@maurges I've changed the secret key to contain `NonZero<SecretScalar<E>>` (instead of `NonZero<Scalar<E>>`), please take a look

File: Cargo.toml (hunk 2 above)
Comment (maurges) [reply Cargo.toml]: Yeah, we should

File: key-export-client/examples/export-request.rs [hunk 3]
```diff
@@ -24,7 +24,7 @@ fn print_export_response() {
 
     let shares = [KeySharePlaintext {
         version: dfns_trusted_dealer_core::version::VersionGuard,
-        index: NonZero::random(&mut rng),
+        index: NonZero::<Scalar<_>>::random(&mut rng),
         secret_share: SecretScalar::<E>::random(&mut rng),
     }]
```
Comment (maurges) [inline key-export-client/examples/export-request.rs]: Should KeySharePlaintext also contain NonZero secret share?

File: Cargo.toml [hunk 4]
```diff
@@ -6,4 +6,8 @@ members = [
   "./key-export-common",
   "./key-export-client",
   "./core",
-]
\ No newline at end of file
+]
+
+[patch.crates-io.key-share]
+git = "https://github.com/dfns/cggmp21"
+branch = "zero-sk"
```
Comment (survived) [inline Cargo.toml]: Now we need to release cggmp21 first

**Comment by maurges** [comment]
Looks good, waiting for cggmp21 now

**Comment by survived** [comment]
@maurges cggmp21 PR has been merged into `m`, so this PR can be merged too.

**Review by maurges — APPROVED** [review]

**Pull request merged**